### PR TITLE
Port to Qt 5.

### PR DESCRIPTION
**Pull request by attente**
Port fcitx-qimpanel to Qt 5. We need this because Qt 4 isn't supported by Mir.

### CODE REVIEW

File: candidate_word.h [hunk 1]
```diff
@@ -20,9 +20,7 @@
 #ifndef __CANDIDATE_WORD_H__
 #define __CANDIDATE_WORD_H__
 
-#include <QDeclarativeView>
-#include <QObject>
-#include <qdeclarative.h>
```
Comment (yuyichao) [inline candidate_word.h]: I thought it is recommanded to include each individual headers in qt5.

Comment (attente) [inline candidate_word.h]: Is there a reason or a link that explains why? I can fix it, but don't know why it's preferred.

Comment (attente) [inline candidate_word.h]: Ok, looking around, I see that it seems to make some projects compile a lot slower. I'll fix this, thanks.

Comment (yuyichao) [inline candidate_word.h]: That's also my guess. I remember seeing it somewhere in a porting guide but could not find it anymore. It also makes it easier to support both Qt4 and Qt5.

**Pull request merged**